CDCgov/prime-reportstream
# 14622: validate a receiver only has one kind of condition filter set

This PR updates the validation of receivers to make sure that only one kind of conditional filter is set

Test Steps:
1. Update a receiver to have both kinds of condition filters (`mappedConditionFilter` and `conditionFilter`)
2. Run the `validate-yaml` CLI command and confirm that you get an error

## Changes
- Adds a new konform rule enforcing that both kinds of condition filters cannot be set

## Checklist

### Testing

### Process

## Linked Issues
#14622 

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
